airflow/providers/slack/hooks/slack_webhook.py

```diff
@@ -19,144 +19,437 @@
+            method attributes and not be required when assign class attributes to message
+            would completely remove.
         """
-        cmd = {}
-
-        if self.channel:
-            cmd['channel'] = self.channel
-        if self.username:
-            cmd['username'] = self.username
-        if self.icon_emoji:
-            cmd['icon_emoji'] = self.icon_emoji
-        if self.icon_url:
-            cmd['icon_url'] = self.icon_url
-        if self.link_names:
-            cmd['link_names'] = 1
-        if self.attachments:
-            cmd['attachments'] = self.attachments
-        if self.blocks:
-            cmd['blocks'] = self.blocks
-
-        cmd['text'] = self.message
-        return json.dumps(cmd)
+        if value is None and name in (
+            "text",
+            "attachments",
+            "blocks",
+            "channel",
+            "username",
+            "icon_emoji",
+            "icon_url",
+            "link_names",
+        ):
+            return getattr(self, name, None)
+
+        return value
+
+    @check_webhook_response
+    def send_dict(self, body: dict[str, Any] | str, *, headers: dict[str, str] | None = None):
+        """
+        Performs a Slack Incoming Webhook request with given JSON data block.
+
+        :param body: JSON data structure, expected dict or JSON-string.
+        :param headers: Request headers for this request.
+        """
+        if isinstance(body, str):
+            try:
+                body = json.loads(body)
+            except json.JSONDecodeError as err:
+                raise AirflowException(
+                    f"Body expected valid JSON string, got {body!r}. Original error:\n * {err}"
+                ) from None
+
+        if not isinstance(body, dict):
+            raise TypeError(f"Body expected dictionary, got {type(body).__name__}.")
+
+        if "link_names" in body:
+            warnings.warn(
+                "`link_names` has no affect, if you want to mention user see: "
+                "https://api.slack.com/reference/surfaces/formatting#mentioning-users",
+                UserWarning,
+                stacklevel=2,
+            )
+        if any(legacy_attr in body for legacy_attr in ("channel", "username", "icon_emoji", "icon_url")):
+            warnings.warn(
+                "You cannot override the default channel (chosen by the user who installed your app), "
+                "username, or icon when you're using Incoming Webhooks to post messages. "
+                "Instead, these values will always inherit from the associated Slack app configuration. "
+                "See: https://api.slack.com/messaging/webhooks#advanced_message_formatting. "
+                "It is possible to change this values only in Legacy Slack Integration Incoming Webhook: "
+                "https://api.slack.com/legacy/custom-integrations/messaging/webhooks#legacy-customizations",
+                UserWarning,
+                stacklevel=2,
+            )
```

Comment:
Another option create separate methods `send_legacy` and `send_text_legacy` which allow send to Legacy Incoming Webhook without any warnings, might be helpful if there is no possible for user change Legacy Incoming Webhook.

Reply:
Is there are good reason why it would not be possible, or just " I had no time to change it?". If that's the latter, then this is no problem to have warnings. Warnings are there to be annoying enough to gently push the user to get rid of the legacy code and make investment in doing so. The only reason for allowing to silence warnings easily is when there migtht be really good reasons and use cases that cannot be handled with the "new way" of doing things.

Reply:
Only one case when user do not have access to create new applications in slack workspace. 
One day `Slack Webhook based on Slack integrations` will be removed anyway so in user perspective better started to migrated from legacy service to 2 successors: Slack API and Slack Incoming Webhook based on API/App

This might be the later feature if someone requested it and better warn user right now always.